File: BoSeonPark/그리디 알고리즘/1931.py
import sys
input = sys.stdin.readline

n = int(input())
timeTable = []
for i in range(n):
    s, e = map(int, input().split(' '))
    timeTable.append((s, e))

timeTable.sort(key = lambda x: (x[1], x[0]))

count = 1
a, b = timeTable[0][0], timeTable[0][1]

for s, e in timeTable[1:]:
    if s < b:
        continue
    else:
        count += 1
        a, b = s, e

print(count)

File: 김하영/Barkingdog_Greedy/1931.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";

const input = fs.readFileSync(filePath).toString().trim().split("\n");
const N = Number(input.shift());
const meetings = input
  .map((str) => str.split(" ").map(Number))
  .sort((a, b) => {
    if (a[1] - b[1] === 0) {
      return a[0] - b[0];
    }
    return a[1] - b[1];
  });

const solution = () => {
  let result = 1;
  let last = 0;
  last = meetings[0][1];
  for (let i = 1; i < meetings.length; i++) {
    if (last <= meetings[i][0]) {
      result++;
      last = meetings[i][1];
    }
  }

  return result;
};

console.log(solution());
